src/components/Toolbar.vue

```diff
@@ -169,11 +173,16 @@ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
 
     </section>
 
-    <section class="modals" v-if="showSaveModal">
-      <save-as-modal
+    <section class="modals">
+      <SaveAsModal
+        v-if="showSaveModal"
         :saveWhat="thingWereSaving"
         :dataToDownload="dataToDownload"
-        :onClose="() => {showSaveModal = false; thingWereSaving = '';}"
+        @close="() => {showSaveModal = false; thingWereSaving = '';}"
+      />
+      <GroundProperties
+        v-else-if="showGroundPropsModal"
+        @close="() => {showGroundPropsModal = false }"
```

Comment:
why do we use an arrow function instead of just a direct assignment here?

Reply:
because I still don't really understand how this works. Seems like if it's a function it gets evaluated when it's bound, and called when the event happens. But if it's an expression it gets evaluated when the event happens? strange...